staker-rewards/rewards.js

```diff
@@ -76,27 +78,38 @@ async function run() {
       decimalSeparator: ".",
     }
 
-    operatorsRewards.forEach((operatorRewards) =>
+    const rewards = {}
+
+    for (const operatorReward of operatorsRewards) {
       console.log(
-        `${operatorRewards.operator}
-           ${operatorRewards.isFraudulent} 
-           ${operatorRewards.keygenCount}
-           ${operatorRewards.keygenFailCount} 
-           ${operatorRewards.keygenSLA} 
-           ${operatorRewards.signatureCount} 
-           ${operatorRewards.signatureFailCount}
-           ${operatorRewards.signatureSLA} 
-           ${operatorRewards.keepStaked.toFormat(format)} 
-           ${operatorRewards.ethBonded.toFormat(format)} 
-           ${operatorRewards.ethUnbonded.toFormat(format)}
-           ${operatorRewards.ethTotal.toFormat(format)} 
-           ${operatorRewards.ethScore.toFormat(noDecimalPlaces, format)} 
-           ${operatorRewards.boost.toFormat(decimalPlaces, format)} 
-           ${operatorRewards.rewardWeight.toFormat(noDecimalPlaces, format)} 
-           ${operatorRewards.totalRewards.toFormat(noDecimalPlaces, format)}
-          `.replace(/\n/g, "\t")
+        `${operatorReward.operator}
+            ${operatorReward.isFraudulent} 
+            ${operatorReward.keygenCount}
+            ${operatorReward.keygenFailCount} 
+            ${operatorReward.keygenSLA} 
+            ${operatorReward.signatureCount} 
+            ${operatorReward.signatureFailCount}
+            ${operatorReward.signatureSLA} 
+            ${operatorReward.keepStaked.toFormat(format)} 
+            ${operatorReward.ethBonded.toFormat(format)} 
+            ${operatorReward.ethUnbonded.toFormat(format)}
+            ${operatorReward.ethTotal.toFormat(format)} 
+            ${operatorReward.ethScore.toFormat(noDecimalPlaces, format)} 
+            ${operatorReward.boost.toFormat(decimalPlaces, format)} 
+            ${operatorReward.rewardWeight.toFormat(noDecimalPlaces, format)} 
+            ${operatorReward.totalRewards.toFormat(noDecimalPlaces, format)}
+            `.replace(/\n/g, "\t")
       )
-    )
+
+      if (operatorReward.totalRewards != 0) {
+        const rewardsBN = new BigNumber(
+          operatorReward.totalRewards.toFormat(noDecimalPlaces, format)
+        )
+        rewards[operatorReward.operator] = rewardsBN.toString(16) // convert BN to hex
+      }
+    }
```

Comment:
`operatorReward.totalRewards` is already a `BigNumber` so:
- the if condition should be either `!operatorReward.totalRewards.isEqualTo(new BigNumber(0))` or `operatorReward.totalRewards.isGreaterThan(new BigDecimal(0))`
- the if body can be simplified to `rewards[operatorReward.operator] = operatorReward.totalRewards.integerValue(BigNumber.ROUND_DOWN).toString(16)`

Reply:
Agree, I used `isZero()` for BigNumber validation. 7b46ae8